### PR TITLE
docs: Add Orc8r v1.6 upgrade docs

### PR DESCRIPTION
**Pull request by hcgatewood**
## Summary

Screen Shot 2021-06-30 at 4 57 35 PM

## Test Plan

Needs to be tested

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: docs/readmes/orc8r/deploy_intro.md [hunk 1]
```diff
@@ -52,22 +52,38 @@ involves the following
...
 
+### v1.6.0
+Verified with Terraform version `0.15.5`.
+
+- `v1.6` [patch branch](https://github.com/magma/magma/tree/v1.6)
+- `github.com/magma/magma//orc8r/cloud/deploy/terraform/orc8r-aws?ref=v1.6`
+Terraform module source
+- `1.5.23` Helm chart version
+- Additional notes
+    - `12.6` PostgreSQL target release, newer versions will likely work as well
```
Comment (karthiksubraveti) [inline docs/readmes/orc8r/deploy_intro.md]: This change(https://github.com/magma/magma/pull/7433) didn't get backported yet. The variables.tf is still pointing to 9.6.

File: docs/readmes/orc8r/upgrade_1_6.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+### Postgres version
+
+[AWS is terminating support for Postgres 9.6 on November 11, 2021](https://aws.amazon.com/blogs/database/upgrading-from-amazon-rds-for-postgresql-version-9-5/), in accordance with the [Postgres versioning policy](https://www.postgresql.org/support/versioning/). Postgres 9.6 was the default Orc8r DB target for the past several releases.
+
+As of Magma release v1.6, the new default Orc8r DB target is Postgres 12.6. *If you are upgrading from a previous release, you will need to handle this change*.
+
+The recommended path is
+1. Pin the Terraform DB engine to your existing version (likely `9.6.15`)
+2. Separately, at a later date, [use Terraform to upgrade to Postgres 12.6](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/db_instance)
```
Comment (karthiksubraveti) [inline docs/readmes/orc8r/upgrade_1_6.md]: Users cannot directly upgradeable from 9.6 -> 12.6. This is the upgrade path
https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_UpgradeDBInstance.PostgreSQL.html
Screen Shot 2021-06-30 at 5 02 22 PM

File: orc8r/cloud/deploy/terraform/orc8r-aws/outputs.tf [hunk 3]
```diff
@@ -11,6 +11,11 @@
 # limitations under the License.
 ################################################################################
 
+output "region" {
```
Comment (karthiksubraveti) [inline orc8r/cloud/deploy/terraform/orc8r-aws/outputs.tf]: Last time I did this ran into some weird issues. I would avoid doing this till we test this change if it is backported to 1.6.

File: docs/readmes/orc8r/upgrade_1_6.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  orc8r_tag = "v1.6.0"
+}
+```
+
+Pull in the module changes
+
+```bash
+terraform init --upgrade
+terraform refresh
+```
```
Comment (karthiksubraveti) [inline docs/readmes/orc8r/upgrade_1_6.md:65]: We added new options in terraform including sns notifications for RDS alerts, setting elasticsearch_disk_threshold we can add that either here or in another document customizing terraform.

**Review by karthiksubraveti — APPROVED** [review]
Mostly LGTM. Left a few comments here.

File: docs/readmes/orc8r/deploy_intro.md [hunk 5]
```diff
@@ -52,22 +52,38 @@ involves the following
...
 
-To target a specific release, checkout the Magma repository's relevant release
-branch when building artifacts. This is also a great place to find relevant
-Terraform values.
+To target a specific release, checkout the Magma repository's relevant release branch when building artifacts. This is also a great place to find relevant Terraform values.
 
 Values for recent Orchestrator releases are summarized below
 
+### v1.6.0
+Verified with Terraform version `0.15.5`.
```
Comment (sudhikan) [inline docs/readmes/orc8r/deploy_intro.md]: I think we are standardizing on `0.15.0` correct? cc: @arunuke

File: docs/readmes/orc8r/deploy_intro.md (hunk 1 above)
Comment (sudhikan) [inline docs/readmes/orc8r/deploy_intro.md]: #7433 needs to be backported to the v1.6 release branch!!! Currently 1.6 is still deploying 9.6.15.

File: docs/readmes/orc8r/deploy_intro.md [hunk 6]
```diff
@@ -52,22 +52,38 @@ involves the following
...
+
+- `v1.6` [patch branch](https://github.com/magma/magma/tree/v1.6)
+- `github.com/magma/magma//orc8r/cloud/deploy/terraform/orc8r-aws?ref=v1.6`
+Terraform module source
+- `1.5.23` Helm chart version
+- Additional notes
+    - `12.6` PostgreSQL target release, newer versions will likely work as well
+
+### v1.5.0
+Verified with Terraform version `0.14.0`. Terraform `0.15.x` is *not* compatible.
```
Comment (sudhikan) [inline docs/readmes/orc8r/deploy_intro.md]: v1.5.0 was standardized on `0.14.5` IIRC.

File: docs/readmes/orc8r/upgrade_1_6.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
+---
+id: upgrade_1_6
+title: Upgrade to v1.6
+hide_title: true
+---
+
+# Upgrade to v1.6
+
+This guide covers upgrading Orchestrator deployments from v1.5 to v1.6. Upgrades which skip versions (e.g. v1.4 to v1.6) are not explicitly documented or supported at this time.
```
Comment (sudhikan) [inline docs/readmes/orc8r/upgrade_1_6.md]: can we say `v1.5.2 to v1.6` please? Release validation did that specifically...

File: orc8r/cloud/deploy/terraform/orc8r-helm-aws/examples/basic/main.tf [hunk 8]
```diff
@@ -12,14 +12,20 @@
...
+  # Change this to pull from GitHub with a specified ref, e.g.
+  # source = "github.com/magma/magma//orc8r/cloud/deploy/terraform/orc8r-aws?ref=v1.6"
   source = "../../../orc8r-aws"
 
   region = "us-west-2"
 
-  orc8r_db_password           = "mypassword" # must be at least 8 characters
+  # If you are performing an upgrade, and are targeting a Postgres version
+  # earlier than current default (12), you must pin to your current version
+  # orc8r_db_engine_version     = "9.6"
```
Comment (sudhikan) [inline orc8r/cloud/deploy/terraform/orc8r-helm-aws/examples/basic/main.tf]: can we change the engine_version to `9.6.15` for consistency?

**Review by sudhikan — APPROVED** [review]
minor comments.

File: docs/readmes/orc8r/deploy_intro.md (hunk 1 above)
Comment (hcgatewood) [reply docs/readmes/orc8r/deploy_intro.md]: Okay good, let's leave it that way actually. I'll update the docs to match

File: docs/readmes/orc8r/upgrade_1_6.md (hunk 2 above)
Comment (hcgatewood) [reply docs/readmes/orc8r/upgrade_1_6.md]: Good info. I view that as implementation detail though, we can figure it out as part of https://github.com/magma/magma/issues/7871

File: orc8r/cloud/deploy/terraform/orc8r-aws/outputs.tf (hunk 3 above)
Comment (hcgatewood) [reply orc8r/cloud/deploy/terraform/orc8r-aws/outputs.tf]: Good to know. Will revert

File: docs/readmes/orc8r/upgrade_1_6.md (hunk 4 above)
Comment (hcgatewood) [reply docs/readmes/orc8r/upgrade_1_6.md:65]: Added notes

File: docs/readmes/orc8r/deploy_intro.md (hunk 5 above)
Comment (hcgatewood) [reply docs/readmes/orc8r/deploy_intro.md]: 👍

File: docs/readmes/orc8r/deploy_intro.md (hunk 1 above)
Comment (hcgatewood) [reply docs/readmes/orc8r/deploy_intro.md]: I didn't realize that change hadn't been backported. I think we should leave it as-is for now (9.6), with a note in the deploy guide that you need to override the engine version and make it e.g. 12.6

File: docs/readmes/orc8r/upgrade_1_6.md (hunk 7 above)
Comment (hcgatewood) [reply docs/readmes/orc8r/upgrade_1_6.md]: Hm. Patch versions are meant to be equivalent from this perspective. From an operator UX perspective, I think we should validate both upgrade paths -- if an operator has v1.5.0 deployed, do they have to upgrade to v1.5.2 first, before they can upgrade to v1.6.0?

From semver docs:

> PATCH version when you make backwards compatible bug fixes

> Consider a version format of X.Y.Z (Major.Minor.Patch). Bug fixes not affecting the API increment the patch version, backwards compatible API additions/changes increment the minor version, and backwards incompatible API changes increment the major version.

Realistically, we're doing semver wrong for our releases. But that aside, I think the operator UX argument is the strongest in favor of retaining the v1.5->v1.6 verbiage

Thoughts?

File: orc8r/cloud/deploy/terraform/orc8r-helm-aws/examples/basic/main.tf (hunk 8 above)
Comment (hcgatewood) [reply orc8r/cloud/deploy/terraform/orc8r-helm-aws/examples/basic/main.tf]: xposting from above: no longer relevant

**Review by electronjoe — APPROVED** [review]

**Pull request merged**